pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -412,6 +421,17 @@ func (a Adapter) createOrUpdateComponent(componentExists bool, ei envinfo.EnvSpe
 		return err
 	}
 
+	var deploymentObjectMeta metav1.ObjectMeta
+	if a.deployment != nil {
+		deploymentObjectMeta = generator.GetObjectMeta(a.deployment.Name, a.Client.Namespace, labels, nil)
+	} else {
+		deploymentName, err := util.NamespaceKubernetesObject(a.ComponentName, a.AppName)
+		if err != nil {
+			return err
+		}
+		deploymentObjectMeta = generator.GetObjectMeta(deploymentName, a.Client.Namespace, labels, nil)
+	}
+
```

Comment:
This is a good candidate for a new method:

```
deploymentObjectMeta, err := a.getDeploymentObjectMeta()
```

will some unit tests on it.

Reply:
Added a new function